app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+      return false;
+    }
+
+    // Hide our deployer pods since it is obvious the deployment is happening
+    // when the new deployment appears.
+    if (label(pod, "openshift.io/deployer-pod-for.name")) {
+      return false;
+    }
+
+    // Hide our build pods since we are already showing details for currently
+    // running or recently run builds under the appropriate areas.
+    if (annotation(pod, "openshift.io/build.name")) {
+      return false;
+    }
+
+    // Hide Jenkins slave pods.
+    if (label(pod, "jenkins") === "slave") {
+      return false;
+    }
+
+    return true;
+  };
+
+  // Group all pods by owner, tracked in the `state.podsByOwnerUID` map.
+  var groupPods = function() {
+    // Make sure the pod owners have loaded before grouping the pods. Otherwise
+    // the pods themselves appear briefly in separate rows as the page loads.
+    if (!overview.pods || !overview.replicationControllers || !overview.replicaSets || !overview.statefulSets) {
+      return;
+    }
+
+    var podOwners = getPodOwners();
+    state.podsByOwnerUID = LabelsService.groupBySelector(overview.pods, podOwners, { key: 'metadata.uid' });
+    overview.monopods = _.filter(state.podsByOwnerUID[''], showMonopod);
+  };
+
+  // Determine if a replication controller is visible, either as part of a
+  // deployment config or a standalone replication controller.
+  var isReplicationControllerVisible = function(replicationController) {
+    if (_.get(replicationController, 'status.replicas')) {
+      return true;
+    }
+    var dcName = annotation(replicationController, 'deploymentConfig');
+    if (!dcName) {
+      return true;
+    }
+    return deploymentIsInProgress(replicationController);
+  };
+
+  // Get the deployment config name for a replication controller by reading the
+  // "openshift.io/deployment-config.name" annotation.
+  var getDeploymentConfigName = function(replicationController) {
+    return annotation(replicationController, 'deploymentConfig');
+  };
+
+  // Group replication controllers by deployment config and filter the visible
+  // replication controllers.
+  // TODO: Handle deleted deployment configs and orphaned replication controllers.
+  var groupReplicationControllers = function() {
+    if (!overview.deploymentConfigs || !overview.replicationControllers) {
+      return;
+    }
+
+    // "Vanilla" replication controllers are those not owned by a deployment config.
+    var vanillaReplicationControllers = [];
+    overview.replicationControllersByDeploymentConfig = {};
+    overview.currentByDeploymentConfig = {};
+    mostRecentByDeploymentConfig = {};
+
+    // Add the replication controllers to a temporary map until we have them all and can sort.
+    var rcByDC = {};
+    _.each(overview.replicationControllers, function(replicationController) {
+      var dcName = getDeploymentConfigName(replicationController) || '';
+      if (!dcName || !overview.deploymentConfigs[dcName]) {
```

Comment:
does this mean if i delete a DC and it doesn't cascade to delete the RCs that I'm going to see a bunch of old RCs showing up that aren't scaled up?